packages/client/types/cors.ts

```diff
@@ -0,0 +1,20 @@
+type CrossOrigin = 'anonymous' | 'use-credentials' | ''
+
+export interface CorsOptions {
+  crossOrigin?: CrossOrigin
+}
+
+/**
+ * In order to cache images in the service worker, CORS has to be enabled.
+ * If CORS is disabled, service worker treats response as opaque and it's not possible to read response headers,
+ * which makes it impossible to cache images.
+ * More information here: https://stackoverflow.com/questions/39109789/what-limitations-apply-to-opaque-responses
+ */
+export const APP_CORS_OPTIONS: CorsOptions = {
+  crossOrigin: 'anonymous'
+}
+
+/**
+ * Using CORS in emails can cause issues with some email clients eg. images not showing in Apple Mail, let's disable it.
+ */
+export const EMAIL_CORS_OPTIONS: CorsOptions = {}
```

Comment:
Some components are used in both emails and app. info in what context component is used is not always available. I decided that adding a separate prop with a cors option is the most viable solution right now, especially when we can have a set of predefined options for emails and app.